packages/flutter/lib/src/gestures/scale.dart

```diff
@@ -36,26 +36,44 @@ enum _ScaleState {
 }
 
 class _PointerPanZoomData {
-  _PointerPanZoomData({
-    required this.focalPoint,
-    required this.scale,
-    required this.rotation
-  });
-  Offset focalPoint;
-  double scale;
-  double rotation;
+  _PointerPanZoomData.fromStartEvent(PointerPanZoomStartEvent event) : _position = event.position, _pan = Offset.zero, _scale = 1, _rotation = 0;
+  _PointerPanZoomData.fromUpdateEvent(PointerPanZoomUpdateEvent event) : _position = event.position, _pan = event.pan, _scale = event.scale, _rotation = event.rotation;
+
+  final Offset _position;
+  final Offset _pan;
+  final double _scale;
+  final double _rotation;
+
+  Offset focalPoint(bool verticalPanShouldActAsZoom) {
+    if (verticalPanShouldActAsZoom) {
+      return _position;
+    }
+    return _position + _pan;
+  }
+
+  double scale(bool verticalPanShouldActAsZoom) {
+    if (verticalPanShouldActAsZoom) {
+      return _scale * math.exp(-_pan.dy / 200);
```

Comment:
For most configurability this could be an `Offset`, and multiplied by `pan.dy` to define your own custom trackpad behaviour. For example, this snippet would be would be `Offset(0, -200)`, but if you had `Offset(200, 0)`, it could scale when scrolling left-right. Then InteractiveViewer would pass through `Offset(0, -scaleFactor)`. But may be of limited use...